### DISCUSSION TITLE
How to calibrate Z-offset properly?

### DISCUSSION THREAD

**Discussion by dosoft** [Guilouz/Klipper-Flsun-Speeder-Pad#85] [Q&A]
Hi, I'm having trouble calibrating the z-offset and would like to know if you can help me out.

Here's what I've done: I ran the z_offset_calibration macro with the sensor plugged in, and it did palpation before lifting up and opening a window with manual adjustment parameters. I then removed the sensor, adjusted the height using a piece of paper, and pressed ACCEPT and SAVE when everything looked OK. However, when I started printing, the nozzle was so close to the print bed that it couldn't extrude any filament. I had to make adjustments on the fly and added 0.75mm to the height. The strange thing is that the first time I calibrated it, the sheet of paper went through fine, so I'm not sure what I'm doing wrong. I've repeated these steps several times and always end up having to fine-tune it adding about 0.75mm, which unfortunately feels a bit dangerous because if my FLSUN SR printer can't push the filament through, it starts to get stuck in the extruder. Do you have any suggestions on how I can achieve a proper calibration without fune-tuning during printing of the first level?

**Comment by dosoft** [comment]
Do you have any ideas? Maybe I'm doing something wrong. Should I run something (macro, command, or other) AFTER disconnecting the sensor but BEFORE adjusting?

Without the speeder pad, when using the original firmware, the usual process is as follows: I run the calibration with the sensor, then remove the sensor and execute a command to move to the zero level (Z=0) before adjusting the head.

Right now, I simply removed the sensor and performed the calibration. Is this correct?

**Reply by zteb** [reply]
I have same issue. About to buy a new FLSUN S1 Pro, swap the mother board and ship it back so I have a machine that works (although not perfect, but it works.)
